src/main/java/de/tum/in/www1/artemis/repository/CompetencyRelationRepository.java

```diff
@@ -23,6 +25,15 @@ public interface CompetencyRelationRepository extends JpaRepository<CompetencyRe
             """)
     Set<CompetencyRelation> findAllByCompetencyId(@Param("competencyId") Long competencyId);
 
+    @Transactional
```

Comment:
Could you try to remove this annotation and see if the test passes. I think it's not necessary here.

Reply:
Removing the annotation actually breaks the tests...